### PR TITLE
Add destination rule feature

### LINKED ISSUE #650: Introduce DestinationRules Views
As [Sebastian](https://github.com/kyma-project/community/blob/main/manifesto/personas.md#sebastian---tech-lead) and [Mathew](https://github.com/kyma-project/community/blob/main/manifesto/personas.md#matthew---cloud-native-software-engineer) I would like to be able to see the list and the details of the [DestinationRules](https://istio.io/latest/docs/reference/config/networking/destination-rule/) that are in My Cluster, and also be able to create and modify existing ones

**Acceptance Criteria**

- [ ] Follow Project DOD
- [ ] This is part of Istio Navigation Grouping  (Gateways are also part of the same group )
- [ ] Listing View is Created. Use the fields  State | Name | Gateways | Hosts | Age - as a starting point
- [ ] Detail View is Create
- [ ] Create Window is available - with yaml editor only
- [ ] Tests are present testing the functionality

### PR DESCRIPTION
**Pull request by wchrapka**
**Description**

Adds Destination Rule list, details, creation and tests.

**Related issue(s)**
Resolves #650

### CODE REVIEW

**Review by akucharska — DISMISSED** [review]

File: core/src/i18n/en.yaml [hunk 1]
```diff
@@ -634,9 +637,65 @@ deployments:
     error: Error
     running: Running
   title: Deployments
-  tooltips:
-    error: At least one of the Pods has failed
-    running: All the Pods are running correctly
+destination-rules:
+  age: Age
+  description: <0>Destination Rule</0> defines policies that apply to traffic intended for a service after routing has occurred.
```
Comment (grego952) [inline core/src/i18n/en.yaml:642]:
```suggestion
  description: <0>Destination Rule</0> specifies rules that apply to traffic intended for a service after routing.
```

**Review by akucharska — DISMISSED** [review]

File: core/src/i18n/en.yaml [hunk 2]
```diff
@@ -634,9 +637,65 @@ deployments:
...
+    consecutive-failures: Consecutive Local Origin Failures
+    consistent-hash: Consistent Hash
+    credential-name: Credential Name
+    distribution: Distribution
+    enabled: Enabled
+    failover: Failover
+    failover-priority: Failover Priority
+    from: From
+    h2-upgrade: H2 Upgrade Policy
+    http-cookie: Http cookie
```
Comment (NHingerl) [inline core/src/i18n/en.yaml:660]:
```suggestion
    http-cookie: HTTP cookie
```

File: core/src/i18n/en.yaml [hunk 3]
```diff
@@ -634,9 +637,65 @@ deployments:
...
+    http-header: Http Header Name
```
Comment (NHingerl) [inline core/src/i18n/en.yaml:661]:
```suggestion
    http-header: HTTP Header Name
```

File: core/src/i18n/en.yaml [hunk 4]
```diff
@@ -634,9 +637,65 @@ deployments:
...
+    http-max: Http2 Max Requests
```
Comment (NHingerl) [inline core/src/i18n/en.yaml:662]:
```suggestion
    http-max: HTTP2 Max Requests
```

File: core/src/i18n/en.yaml [hunk 5]
```diff
@@ -634,9 +637,65 @@ deployments:
...
+    http-pending: Http1 Max Pending Requests
```
Comment (NHingerl) [inline core/src/i18n/en.yaml:663]:
```suggestion
    http-pending: HTTP1 Max Pending Requests
```

File: core/src/i18n/en.yaml [hunk 6]
```diff
@@ -634,9 +637,65 @@ deployments:
...
+    http-query-parameter: Http Query Parameter Name
```
Comment (NHingerl) [inline core/src/i18n/en.yaml:664]:
```suggestion
    http-query-parameter: HTTP Query Parameter Name
```

File: core/src/i18n/en.yaml [hunk 7]
```diff
@@ -634,9 +637,65 @@ deployments:
...
+    http-settings: HTTP Connection Pool Settings
+    idle-timeout: Idle Timeout
+    interval: Interval
+    lb-settings: Locality Lb Settings
```
Comment (NHingerl) [inline core/src/i18n/en.yaml:668]: Is Lb "Load Balancer"? Then I'd suggest 
 ```suggestion
    lb-settings: Locality LB Settings
```

File: core/src/i18n/en.yaml [hunk 8]
```diff
@@ -634,9 +637,65 @@ deployments:
...
+    subset: Subset
+    subsets: Subsets
+    tcp-alive: TCP Keep Alive
+    tcp-settings: TCP Connection Pool Settings
+    time: Time
+    tls-settings: Client TLS Settings
+    to: To
+    traffic-policy: Traffic Policy
+    ttl: TTL
+  exported-to-namespaces: Exported to namespaces
```
Comment (NHingerl) [inline core/src/i18n/en.yaml:694]:
```suggestion
  exported-to-namespaces: Exported to Namespaces
```

File: core/src/i18n/en.yaml [hunk 9]
```diff
@@ -634,9 +637,65 @@ deployments:
...
+  description: <0>Destination Rule</0> specifies rules that apply to traffic intended for a service after routing.
+  details:
+    alt-names: Subject Alt Names
+    base-ejection: Base Ejection Time
+    ca-certificates: Ca Certificates
+    client-certificate: Client Certificate
+    client-protocol: Use Client Protocol
+    connect-timeout: Connect Timeout
+    consecutive-5xx: Consecutive 5xx Errors
+    consecutive-failures: Consecutive Local Origin Failures
```
[the remark this answers is not in the document]
Comment (wchrapka) [reply core/src/i18n/en.yaml:663]: Yes, by the API:

> Determines whether to distinguish local origin failures from external errors. If set to true consecutivelocalorigin_failure is taken into account for outlier detection calculations.

**Review by akucharska — APPROVED** [review]

File: core/src/i18n/en.yaml (hunk 9 above)
[the remark this answers is not in the document]
Comment (NHingerl) [reply core/src/i18n/en.yaml:663]: Perfect, thank you!

**Review by NHingerl — APPROVED** [review]

**Pull request merged**